### ISSUE TITLE
[bug] Requirement options with None not working

### ISSUE THREAD

**Issue by gummif** [conan-io/conan#13717]
### Environment details

* Operating System+version: Ubuntu 20.04.5 LTS
* Compiler+version: gcc 9
* Conan version: 2.0.4
* Python version: 3.8.10

### Steps to reproduce

With conanfile.py
```python
from conan import ConanFile

class Tester(ConanFile):
    settings = "os", "compiler", "build_type", "arch"
    generators = "CMakeToolchain", "CMakeDeps"
    requires = "zeromq/4.3.4#c7733f877a59582f2e77334b345b9a00"
    default_options = {
        "zeromq/*:encryption": None
    }
```
Execute
```
conan install . --build missing
```
Which builds libsodium (the default value in zeromq), but should not build anything (zeromq has options "encryption": [None, "libsodium", "tweetnacl"]).

Looks like None is treated as not set, but not a value passed to the dependency (if I e.g. set `"zeromq/*:shared": None` which is not a valid option for `shared` then I get no error).

### Logs

Calling install again:
$ conan install . --build missing

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu14
compiler.libcxx=libstdc++11
compiler.version=9
os=Linux

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu14
compiler.libcxx=libstdc++11
compiler.version=9
os=Linux

======== Computing dependency graph ========
Graph root
    conanfile.py: /home/vagrant/test/conanfile.py
Requirements
    libsodium/1.0.18#9e310e52ed60084484c5f640ab88883a - Cache
    zeromq/4.3.4#c7733f877a59582f2e77334b345b9a00 - Cache

======== Computing necessary packages ========
Requirements
    libsodium/1.0.18#9e310e52ed60084484c5f640ab88883a:b19064868c3be0252375d2b863fb2b8f03002e2e#a47e6ebfb903934d8f96391e49b5de99 - Cache
    zeromq/4.3.4#c7733f877a59582f2e77334b345b9a00:118804eb9ba279377ea7c212ed51e28968d96038#9eab2a57ce65417ff70829bd854a0db2 - Cache

======== Installing packages ========
libsodium/1.0.18: Already installed! (1 of 2)
zeromq/4.3.4: Already installed! (2 of 2)
WARN: Usage of deprecated Conan 1.X features that will be removed in Conan 2.X:
WARN:     'cpp_info.names' used in: zeromq/4.3.4
WARN:     'cpp_info.build_modules' used in: zeromq/4.3.4

======== Finalizing install (deploy, generators) ========
conanfile.py: Writing generators to /home/vagrant/test
conanfile.py: Generator 'CMakeDeps' calling 'generate()'
conanfile.py: Generator 'CMakeToolchain' calling 'generate()'
conanfile.py: CMakeToolchain generated: conan_toolchain.cmake
conanfile.py: Preset 'conan-release' added to CMakePresets.json. Invoke it manually using 'cmake --preset conan-release' if using CMake>=3.23
conanfile.py: If your CMake version is not compatible with CMakePresets (<3.23) call cmake like: 'cmake <path> -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE=/home/vagrant/test/conan_toolchain.cmake -DCMAKE_POLICY_DEFAULT_CMP0091=NEW -DCMAKE_BUILD_TYPE=Release'
conanfile.py: CMakeToolchain generated: CMakePresets.json
conanfile.py: Generating aggregated env files
conanfile.py: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
Install finished successfully

**Comment by memsharded** [comment]
Hi @gummif 

thanks for your report.

You are right about your observation:

> Looks like None is treated as not set, 

In Conan 2.0, the Python ``None`` is the same as "not defined". When defined in an option like ``myoption: [None, True, False]`` it doesn't mean that it can take the ``None`` value, it means that it can take the True and False values, or be left undefined. So this seems to be working as expected.

> zeromq has options "encryption": [None, "libsodium", "tweetnacl"]

This seems the gap, it is in the recipe, that defines ``None`` as an option, but then defaults to ``libsodium``. This is not possible (there were some ugly cracks in 1.X that could have allowed this to not be raised, because there are automatic equivalences between ``"None"`` as string and Python ``None``.

I'd recommend moving this to ``conan-center-index`` repo, as it should be fixed in the recipe.

**Comment by gummif** [comment]
Ok I see, thanks. So the fix would be to add an option like "none" that would be treated like `None` was in conan v1 (and I guess keeping `None` around for v1 compatibility)?

**Comment by memsharded** [comment]
yes, exactly if there is some option value that has some semantics different than "this is not defined"; then a string with ``"none"`` or the like seems the way to go, and indeed keeping the ``None`` for compatibility could help to not break.
